core/symbol/casecmp_spec.rb

```diff
@@ -72,3 +72,62 @@
     :abc.casecmp(obj).should be_nil
   end
 end
+
+ruby_version_is "2.4" do
+  describe 'Symbol#casecmp?' do
+    it "compares symbols without regard to case" do
+      :abcdef.casecmp?(:abcde).should == false
+      :aBcDeF.casecmp?(:abcdef).should == true
+      :abcdef.casecmp?(:abcdefg).should == false
+      :abcdef.casecmp?(:ABCDEF).should == true
+    end
+
+    it "doesn't consider non-ascii characters equal that aren't" do
+      # -- Latin-1 --
+      upper_a_tilde  = :"\xC3"
+      upper_a_umlaut = :"\xC4"
+      lower_a_tilde  = :"\xE3"
+      lower_a_umlaut = :"\xE4"
+
+      lower_a_tilde.casecmp(lower_a_umlaut).should_not == true
+      lower_a_umlaut.casecmp(lower_a_tilde).should_not == true
+      upper_a_tilde.casecmp(upper_a_umlaut).should_not == true
+      upper_a_umlaut.casecmp(upper_a_tilde).should_not == true
+
+      # -- UTF-8 --
+      upper_a_tilde  = :"\xC3\x83"
+      upper_a_umlaut = :"\xC3\x84"
+      lower_a_tilde  = :"\xC3\xA3"
+      lower_a_umlaut = :"\xC3\xA4"
+
+      lower_a_tilde.casecmp(lower_a_umlaut).should_not == true
+      lower_a_umlaut.casecmp(lower_a_tilde).should_not == true
+      upper_a_tilde.casecmp(upper_a_umlaut).should_not == true
+      upper_a_umlaut.casecmp(upper_a_tilde).should_not == true
+    end
+
+    it "doesn't do case mapping for non-ascii characters" do
+      # -- Latin-1 --
+      upper_a_tilde  = :"\xC3"
+      upper_a_umlaut = :"\xC4"
+      lower_a_tilde  = :"\xE3"
+      lower_a_umlaut = :"\xE4"
+
+      upper_a_tilde.casecmp(lower_a_tilde).should == false
```

Comment:
This spec fails, I suspect due to an encoding issue, but I'd appreciate a second opinion on that :slightly_smiling_face:

Reply:
Yes, it's due to the magic encoding comment at the top of the file.
Could you change it to UTF-8 like in https://github.com/ruby/spec/pull/474/commits/24dfde2fcde97634ba7e999e62914b28b8b55fa3?

Reply:
:+1: will do

Reply:
Unfortunately doing this breaks other specs in this file that test Latin-1 characters. How would you like me to proceed with this?

Reply:
If you can easily fix those specs, then that would be the best (it's fine in this PR) :)
Otherwise, I can take a look and make the encoding change first.

Reply:
Mmh, it's a little bit tricky, it seems MRI does not accept Symbol with invalid characters.
I have a fix locally, will push it to master and then you can rebase on top.

Reply:
Done in 6f182f9700aace3de4448bade9f463577a98a6cd, please rebase with:
```bash
git fetch
git rebase origin/master
git push -f
```
Then the CI should pass :smile:

Reply:
Ah, I just got this working. I realised I was calling `casecmp` instead of `casecmp?` :man_facepalming: I'll rebase and see where that gets me

Reply:
:smile: